CodePush.m

```diff
@@ -468,6 +528,60 @@ - (void)savePendingUpdate:(NSString *)packageHash
 }
 
 /*
+ * This method is checks if a new status update exists (new version was installed, 
+ * or an update failed) and return its details (version label, status).
+ */
+RCT_EXPORT_METHOD(getNewStatusReport:(RCTPromiseResolveBlock)resolve
+                            rejecter:(RCTPromiseRejectBlock)reject)
+{
+    if (didRollback) {
+        // Check if there was a rollback that was not yet reported
+        NSUserDefaults *preferences = [NSUserDefaults standardUserDefaults];
+        NSMutableArray *failedUpdates = [preferences objectForKey:FailedUpdatesKey];
+        if (failedUpdates) {
+            NSDictionary *lastFailedPackage = [failedUpdates lastObject];
+            if (lastFailedPackage) {
+                NSString *lastFailedPackageIdentifier = [self getPackageStatusReportIdentifier:lastFailedPackage];
+                if (lastFailedPackageIdentifier && [self isDeploymentStatusNotYetReported:lastFailedPackageIdentifier]) {
+                    [self recordDeploymentStatusReported:lastFailedPackageIdentifier];
+                    resolve(@{ @"package": lastFailedPackage, @"status": DeploymentFailed });
+                    return;
+                }
+            }
+        }
+    } else if (_isFirstRunAfterUpdate) {
+        // Check if the current CodePush package has been reported
+        NSError *error;
+        NSDictionary* currentPackage = [CodePushPackage getCurrentPackage:&error];
+        if (currentPackage) {
+            NSString* currentPackageIdentifier = [self getPackageStatusReportIdentifier:currentPackage];
+            if (currentPackageIdentifier && [self isDeploymentStatusNotYetReported:currentPackageIdentifier]) {
+                [self recordDeploymentStatusReported:currentPackageIdentifier];
+                resolve(@{ @"package": currentPackage, @"status": DeploymentSucceeded });
+                return;
+            }
+        }
+    } else {
+        if (isRunningBinaryVersion) {
```

Comment:
This condition seems to run every time I call `notifyApplicationReady`, so I'm not sure whether the report is being saved or not.